src/parser.rs

```diff
@@ -442,10 +447,58 @@ struct SatsPack {
     /// BeiDou: 12 + 3 IGSO + 3 GEO
     /// Galileo: 12
     /// => 58 total Satellites => max 15 rows of data
+    #[cfg_attr(feature = "serde", serde(serialize_with = "serialize_deque"))]
+    #[cfg_attr(feature = "serde", serde(deserialize_with = "deserialize_deque"))]
     data: Deque<Vec<Option<Satellite>, 4>, 15>,
     max_len: usize,
 }
 
+#[cfg(feature = "serde")]
+fn serialize_deque<S>(v: &Deque<Vec<Option<Satellite>, 4>, 15>, s: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let mut seq = s.serialize_seq(Some(15))?;
+    for e in v.iter() {
+        seq.serialize_element(e)?;
+    }
+    seq.end()
+}
+
+#[cfg(feature = "serde")]
+struct DequeVisitor;
+
+#[cfg(feature = "serde")]
+impl<'de> Visitor<'de> for DequeVisitor {
+    type Value = Deque<Vec<Option<Satellite>, 4>, 15>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("deque")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'de>,
+    {
+        let mut deq: Deque<Vec<Option<Satellite>, 4>, 15> = Deque::new();
+
+        while let Some(v) = seq.next_element()? {
+            deq.push_back(v).expect("Cannot deserialize");
```

Comment:
panic is inappropriate, you can early return with an Err case containing an error created using this [trait](https://docs.rs/serde/latest/serde/de/trait.Error.html).

[Example](https://serde.rs/deserialize-struct.html#manually-implementing-deserialize-for-a-struct)